server/athenian/api/align/mutations/goals.py

```diff
@@ -22,44 +19,35 @@ async def resolve_create_goal(
     info: GraphQLResolveInfo,
     accountId: int,
     input: Dict[str, Any],
-):
+) -> dict:
     """Create a Goal."""
-    parse_result = _parse_create_goal_input(input, accountId)
+    creation_info = _parse_create_goal_input(input, accountId)
 
     async with info.context.sdb.connection() as sdb_conn:
         async with sdb_conn.transaction():
-            await _parse_result_db_validation(parse_result, sdb_conn)
-            new_goal_id = await _insert_db_goal(parse_result, sdb_conn)
+            new_goal_id = await insert_goal(creation_info, sdb_conn)
 
     # TODO: return the complete response
     return {"goal": {"id": new_goal_id}}
 
 
-class GoalMutationError(ResponseError):
-    """An error during a goal mutation handling."""
-
-    def __init__(self, text):
-        """Init the GoalMutationError."""
-        wrapped_error = GenericError(
-            type="/errors/align/GoalMutationError",
-            status=HTTPStatus.BAD_REQUEST,
-            detail=text,
-            title="Goal mutation error",
-        )
-        super().__init__(wrapped_error)
-
-
-@dataclass(frozen=True)
-class _CreateGoalInputParseResult:
-    goal: Goal
-    team_goals: Sequence[TeamGoal]
+@mutation.field("removeGoal")
+async def resolve_remove_goal(
+    _: Any,
+    info: GraphQLResolveInfo,
+    accountId: int,
+    id: int,
+) -> dict:
+    """Remove a Goal and referring TeamGoal-s."""
+    async with info.context.sdb.connection() as sdb_conn:
+        # foreign_keys support is required in sqlite to cascade delete TeamGoal-s
+        async with enable_sqlite_foreign_keys(info.context.sdb, sdb_conn):
+            await delete_goal(accountId, id, sdb_conn)
+    return {"success": True}
```

Comment:
I am not a big fan of hardcoding keys in responses. Can we leverage the GraphQL model like we do in OpenAPI?
- Create the response model
- Return the response model
- Something else unpacks the model back to dict

Reply:
done